packages/fuels-accounts/src/provider.rs

```diff
@@ -188,7 +203,17 @@ impl Provider {
     /// Sends a transaction to the underlying Provider's client.
     pub async fn send_transaction_and_await_commit<T: Transaction>(&self, tx: T) -> Result<TxId> {
         let tx_id = self.send_transaction(tx.clone()).await?;
-        self.client.await_transaction_commit(&tx_id).await?;
+        let _status = self.client.await_transaction_commit(&tx_id).await?;
+
+        #[cfg(feature = "coin-cache")]
+        {
+            if matches!(
+                _status,
+                TransactionStatus::SqueezedOut { .. } | TransactionStatus::Failure { .. }
+            ) {
+                self.cache.lock().await.remove_items(tx.used_coins())
+            }
+        }
```

Comment:
Is there any way we can test this at this layer (as opposed to only testing the cache implementation)? E.g., forcing a tx failure and checking the cached utxo. It doesn't have to be done in this PR, though. But it would be nice to have a proper check at this level.